java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowReader.java

```diff
@@ -243,6 +248,8 @@ protected void loadDictionary(ArrowDictionaryBatch dictionaryBatch) {
       }
       return;
     }
+    // TODO - If the super class is a file reader it may be good to throw an exception here
+    // the dictionary failed to satisfy the spec (i.e. being a replacement dictionary)
```

Comment:
It would be better if we could create an issue for this.

Reply:
Ack, do you leave TODOs in or just remove them and create the issue?

Reply:
Can we enforce this in the reader (instead of a todo?)